client/src/app/queries/waves.ts

```diff
@@ -0,0 +1,77 @@
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { Wave } from "@app/api/models";
+import { AxiosError } from "axios";
+
+// TODO: Implement delete api
+
+let deleteWave: any;
+
+export interface IWaveFetchState {
+  waves: Wave[];
+  isFetching: boolean;
+  fetchError: any;
+  refetch: any;
+}
+
+export const WavesQueryKey = "stakeholders";
+
+export const useFetchWaves = (): IWaveFetchState => {
+  const { data, isLoading, error, refetch } = useQuery(
+    WavesQueryKey,
+    async () => [],
+    // async () => (await getWaves()).data,
+    {
+      onError: (error) => console.log("error, ", error),
+    }
+  );
+  return {
+    waves: [
+      {
+        name: "wave1",
+        id: 0,
+        startDate: "2018-02-03 22:15:01",
+        endDate: "2020-01-03 22:15:014",
+        applications: [{ name: "app1" }, { name: "app2" }],
+        stakeholders: [{ name: "sh1", email: "aaakwd@aol.com" }],
+        status: "n/a",
+      },
+      {
+        name: "wave2",
+        id: 2,
+        startDate: "2019-03-03 22:15:01",
+        endDate: "2020-01-03 22:15:014",
+        applications: [{ name: "app3" }, { name: "app4" }],
+        stakeholders: [{ name: "sh1", email: "aaakwd@aol.com" }],
+        status: "n/a",
+      },
+    ],
+
+    // waves: data || [],
+    isFetching: isLoading,
+    fetchError: error,
+    refetch,
+  };
+};
+
+export const useDeleteWaveMutation = (
+  onSuccess: (res: any) => void,
+  onError: (err: AxiosError) => void
```

Comment:
This can be replaced with `onError: (err: Error | null) => void` 

And the OnError below could be like: 
```
    onError: (err: Error) => {
      onError && onError(err);
    },
```

The reason is about the coming api/rest part.

The details are https://github.com/konveyor/tackle2-ui/issues/607

Reply:
Since this is the onError callback that we are running when an error occurs, we know that the error will always be an AxiosError (since we are using axios). 

and 
```onError && onError ```
shouldn't be needed since we are expecting to always have an onError callback in place for handling notifications/sideEffects, right?

Reply:
Maybe let's discuss that when we're consolidating our requests ? 
For now either leave it or remove that code since there isn't any real API requests yet.